lib/AST/ASTPrinter.cpp

```diff
@@ -4275,8 +4282,21 @@ void PrintAST::visitSequenceExpr(SequenceExpr *expr) {
 void PrintAST::visitSuperRefExpr(SuperRefExpr *expr) {
 }
 
+void PrintAST::printBaseExprForTypeAccess(Expr *expr) {
+  if (auto *tExpr = dyn_cast<TypeExpr>(expr)) {
+    if (auto metaType = tExpr->getType()->castTo<AnyMetatypeType>()) {
+      // Don't print `.Type` for an expr.
+      printType(metaType->getInstanceType());
```

Comment:
I believe `TypeExpr` has a `getInstanceType()` method you could use here. That being said, it kind of seems like this should be the default printing logic for `TypeExpr` anyway (in `visitTypeExpr`), what do you think?

Reply:
Ah, yes, putting this in `visitTypeExpr` is much nicer and works just as well. Thanks!